src/osd/PrimaryLogPG.cc

```diff
@@ -2044,6 +2044,15 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
     }
   }
 
+  // check for op with rwordered and rebalance reads
+  if (m->has_flag(CEPH_OSD_FLAG_BALANCE_READS) &&
+      op->rwordered()) {
```

Comment:
What about a `rwordered` localized read after the rework? My understanding of the `CEPH_OSD_FLAG_RWORDERED` is that an operation should be handled basically like a write (in terms of ordering) which means **it needs to happen on primary**.

I don't see a reason why a balanced (handling OSD picked up randomly) read is different from localized (handled by _closest_ OSD) read.

Reply:
mm right, it may or may not be the primary, I'll add a check for localized flag as well.